txpool/pool.go

```diff
@@ -628,6 +635,17 @@ func (p *TxPool) Best(n uint16, txs *types.TxsRlp, tx kv.Tx, onTopOf uint64) (bo
 				toRemove = append(toRemove, mt)
 				continue
 			}
+
+			// make sure we have enough gas in the caller to add this transaction.
+			// not an exact science using intrinsic gas but as close as we could hope for at
+			// this stage
+			intrinsicGas, _ := CalcIntrinsicGas(uint64(mt.Tx.DataLen), uint64(mt.Tx.DataNonZeroLen), nil, mt.Tx.Creation, true, true)
+			if intrinsicGas > availableGas {
+				// we might find another TX with a low enough intrinsic gas to include so carry on
+				continue
+			}
+			availableGas -= intrinsicGas
```

Comment:
uint underflow?

Reply:
Would the line above this not save us from that?  Checking that the intrinsic gas is not higher than available gas

Reply:
maybe... until somebody add new condition to line above...

Reply:
I've dropped a quick check around this statement explicitly now to check for an underflow before changing the availableGas.  This is in the new PR #755